lit_gpt/lora.py

```diff
@@ -149,16 +149,15 @@ def T(self, x: torch.Tensor) -> torch.Tensor:
 
     def merge(self):
         """Merges the LoRA weights into the full-rank weights (W = W + delta_W)."""
-
         if self.r > 0 and not self.merged:
             # Merge the weights and mark it
-            self.weight.data += self.T(self.lora_B @ self.lora_A) * self.scaling
+            self.linear.weight.data += self.T(self.lora_B @ self.lora_A) * self.scaling
             self.merged = True
 
     def forward(self, x: torch.Tensor):
         # if weights are merged or rank is less or equal to zero (LoRA is disabled) - it's only a regular nn.Linear forward pass;
         # otherwise in addition do the forward pass with LoRA weights and add it's output to the output from pretrained weights
-        result = F.linear(x, self.T(self.weight), bias=self.bias)
+        result = self.linear(x)
```

Comment:
`self.T` is a conditional method, that means that if `self.fan_in_fan_out` is True then this might not work.
By default `self.fan_in_fan_out` is False so maybe that is why you haven't seen any errors.

Reply:
Good catch. I can't think of a solution to that, so I'll just drop the argument since the project doesn't use it